src/hotspot/share/gc/shared/referenceProcessor.cpp

```diff
@@ -771,16 +793,31 @@ void ReferenceProcessor::balance_queues(DiscoveredList ref_lists[])
 #endif
 }
 
-void ReferenceProcessor::process_soft_ref_reconsider(BoolObjectClosure* is_alive,
-                                                     OopClosure* keep_alive,
-                                                     VoidClosure* complete_gc,
-                                                     AbstractRefProcTaskExecutor* task_executor,
-                                                     ReferenceProcessorPhaseTimes* phase_times) {
-  assert(!processing_is_mt() || task_executor != NULL, "Task executor must not be NULL when mt processing is set.");
+void ReferenceProcessor::run_task(AbstractGangTask& task, AbstractClosureContext& closure_context, bool marks_oops_alive) {
+  WorkGang* gang = Universe::heap()->safepoint_workers();
+  assert(gang != NULL || !processing_is_mt(), "can not dispatch multi threaded without a work gang");
+  log_debug(gc, ref)("ReferenceProcessor::execute queues: %d, %s, marks_oops_alive: %s",
+    num_queues(), processing_is_mt() ? "ThreadModel::Multi" : "ThreadModel::Single", marks_oops_alive?"true":"false");
+  closure_context.prepare_run_task(num_queues(), processing_is_mt() ? ThreadModel::Multi : ThreadModel::Single, marks_oops_alive);
+  if (gang != NULL && processing_is_mt()) {
+    assert(gang->active_workers() >= num_queues(),
+           "Ergonomically chosen workers(%u) should be less than or equal to active workers(%u)",
+           num_queues(), gang->active_workers());
+    gang->run_task(&task, num_queues());
+  } else {
+    log_debug(gc, ref)("Serial loop: %d\n", _max_num_queues);
```

Comment:
This log message seems to be a debugging message, not offering a lot information. Please remove.

Reply:
Okay.